lua/wikis/commons/Infobox/Extension/TeamHistoryAuto.lua

```diff
@@ -0,0 +1,367 @@
+---
+-- @Liquipedia
+-- page=Module:Infobox/Extension/TeamHistoryAuto
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+--remaining issue: player joins/leaves same team with multiple roles on different dates (e.g. MarioMe LoL wiki)
+
+local Class = require('Module:Class')
+local Lua = require('Module:Lua')
+
+local Array = Lua.import('Module:Array')
+local DateExt = Lua.import('Module:Date/Ext')
+local FnUtil = Lua.import('Module:FnUtil')
+local Info = Lua.import('Module:Info', {loadData = true})
+local Json = Lua.import('Module:Json')
+local Logic = Lua.import('Module:Logic')
+local Namespace = Lua.import('Module:Namespace')
+local Roles = Lua.import('Module:Roles')
+local Table = Lua.import('Module:Table')
+local TransferModel = Lua.import('Module:Transfer/Model')
+local TransferRef = Lua.import('Module:Transfer/References')
+
+local Link = Lua.import('Module:Widget/Basic/Link')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local Abbr = HtmlWidgets.Abbr
+local Fragment = HtmlWidgets.Fragment
+local Span = HtmlWidgets.Span
+local Tbl = HtmlWidgets.Table
+local Td = HtmlWidgets.Td
+local Th = HtmlWidgets.Th
+local Tr = HtmlWidgets.Tr
+local WidgetUtil = Lua.import('Module:Widget/Util')
+
+local Condition = require('Module:Condition')
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local SPECIAL_ROLES = {
+	'Retired',
+	'Retirement',
+	'Military',
+	'Banned',
+	'Producer',
+	'Caster',
+	'Admin',
+	'Observer',
+	'Host',
+	'Talent',
+	'League Operator',
+	'Inactive'
+}
+local LOAN = 'Loan'
+
+local POSITION_ICON_DATA = Lua.requireIfExists('Module:PositionIcon/data', {loadData = true})
+
+-- todo: decide what to do with those ...
+local NOT_YET_IN_ROLES_DATA = {
+	['coach/analyst'] = {display = 'Coach/Analyst', children = {'C./A.'}},
+	['coach and analyst'] = {display = 'Coach/Analyst', children = {'C./A.'}},
+	['overall coach'] = {display = 'Overall Coach', children = {'OC.'}},
+	['manager and analyst'] = {display = 'Manager/Analyst', children = {'M./A.'}},
+	['manager/analyst'] = {display = 'Manager/Analyst', children = {'M./A.'}},
+	['general manager'] = {display = 'General Manager', children = {'GM.'}},
+	['assistant general manager'] = {display = 'Assistant General Manager', children = {'AGM.'}},
+	['team manager'] = {display = 'Team Manager', children = {'TM.'}},
+	['assistant team manager'] = {display = 'Assistant Team Manager', children = {'ATM.'}},
+	substitute = {display = 'Substitute', children = {'Sub.'}},
+	inactive = {display = 'Inactive', children = {'Ia.'}},
+	['training advisor'] = {display = 'Training Advisor', children = {'TA.'}},
+	['founder & training director'] = {display = 'Founder & Training Director', children = {'F. & TD.'}},
+}
```

Comment:
@Rathoz 
what should we do with these?
they are (currently) not supported in standardized role data